plugins/catalog-backend/config.d.ts

```diff
@@ -17,226 +17,370 @@
+     * Rules to apply to all catalog entities, from any location.
+     *
+     * An undefined list of matchers means match all, an empty list of
+     * matchers means match none.
+     *
+     * This is commonly used to put in what amounts to a whitelist of kinds
+     * that regular users of Backstage are permitted to register locations
+     * for. This can be used to stop them from registering yaml files
+     * describing for example a Group entity called "admin" that they make
+     * themselves members of, or similar.
      */
     rules?: Array<{
       /**
-       * attr: 'allow' - accepts an array of string. e.g. [Component, API, Group, User, Template, Location]
+       * Allow entities of these particular kinds.
+       *
+       * E.g. ["Component", "API", "Template", "Location"]
        */
       allow: Array<string>;
+    }>;
+
+    /**
+     * A set of static locations that the catalog shall always keep itself
+     * up-to-date with. This is commonly used for large, permanent integrations
+     * that are defined by the Backstage operators at an organization, rather
+     * than individual things that users register dynamically.
+     *
+     * These have (optional) rules of their own. These override what the global
+     * rules above specify. This way, you can prevent everybody from register
+     * e.g. User and Group entities, except for one or a few static locations
+     * that have those two kinds explicitly allowed.
+     *
+     * For example:
+     *
+     * ```yaml
+     * rules:
+     *   - allow: [Component, API, Template, Location]
+     * locations:
+     *   - type: url
+     *     target: https://github.com/org/repo/blob/master/users.yaml
+     *     rules:
+     *       - allow: [User, Group]
+     *   - type: url
+     *     target: https://github.com/org/repo/blob/master/systems.yaml
+     *     rules:
+     *       - allow: [System]
+     * ```
+     */
+    locations?: Array<{
       /**
-       * opt-attr: 'locations' - accepts an array of key, value pair objects.
-       * keys: 'target', 'type'
-       * e.g.[{type: url, target: https://github.com}]
+       * The type of location, e.g. "url".
+       */
+      type: string;
+      /**
+       * The target URL of the location, e.g.
+       * "https://github.com/org/repo/blob/master/users.yaml".
+       */
+      target: string;
+      /**
+       * Optional extra rules that apply to this particular location.
        *
+       * These override the global rules above.
        */
-      locations?: Array<{ [key: string]: string }>;
+      rules?: Array<{
+        /**
+         * Allow entities of these particular kinds.
+         *
+         * E.g. ["Group", "User"]
+         */
+        allow: Array<string>;
+      }>;
     }>;
+
     /**
-     * List of processor options and attributes
+     * List of processor-specific options and attributes
      */
     processors?: {
```

Comment:
@Rugvip, @freben - I'm curious, will it be possible to extend keys like `processors` to add custom ones in the backend. We've started added some custom keys like `catalog.refresh.delay` and have some custom processors that have config settings as well. Might be good to have some guidance around this.